clang/lib/Driver/Driver.cpp

```diff
@@ -4359,9 +4374,13 @@ void Driver::BuildJobs(Compilation &C) const {
 
   // If we have more than one job, then disable integrated-cc1 for now. Do this
   // also when we need to report process execution statistics.
-  if (C.getJobs().size() > 1 || CCPrintProcessStats)
-    for (auto &J : C.getJobs())
-      J.InProcess = false;
+  if (C.getJobs().size() > 1 || CCPrintProcessStats) {
+    if (C.getJobs().size() != 2 ||
+        !isa<DepscanJobAction>(C.getJobs().begin()->getSource())) {
```

Comment:
Worth a comment for why depscan gets special handling here

Reply:
It doesn't seem like `CCPrintProcessStats` should trigger out-of-process `-cc1depscan`. Can you confirm why/whether this makes sense?

Reply:
> It doesn't seem like `CCPrintProcessStats` should trigger out-of-process `-cc1depscan`. Can you confirm why/whether this makes sense?

Looking more closely, I think this is fully consistent with the existing code for `CCPrintProcessStats`. I think we can mark this as "resolved" again 👍 .